src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -347,7 +375,7 @@ static void bus_reset(uint8_t rhport) {
   dwc2->grxfsiz = calc_grxfsiz(64, ep_count);
 
   // Setup the control endpoint 0
-  _allocated_fifo_words_tx = 16;
+  _allocated_fifo_words_tx = 32;
```

Comment:
It seems like EP0's IN FIFO can't be located at the end of RAM, otherwise IN packet is corrupted like above.

Reply:
For U5 (maybe also ESP32-S3 who failed HIL) I've to reserve an excessive amount of space like 90 words in order to make it work, otherwise packet ending will be corrupted like my 1st comment.

@hathach I doubt it has something to do with new feature like `dma_dynamic`, ST hasn't update their comment for U5 in HAL driver:
```
When DMA is used 3n * FIFO locations should be reserved for internal DMA registers */
```
I think I need help from somebody knowing dwc2 internal structures...